src/main/java/DukeFile.java

```diff
@@ -0,0 +1,111 @@
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+
+public class DukeFile {
+    private String path;
+
+    public DukeFile(String path) {
+        // Splits pathname into relative path and filename
+        // Creates missing dir/ file
+        this.path = path;
+        String filename = path.substring(path.lastIndexOf("/") + 1);
+        String dir = path.split(filename)[0];
+        File d = new File(dir);
+        File f = new File(path);
+        if (!d.exists()) {
+            d.mkdirs();
```

Comment:
I think "mkdirs" should be "mkDirs" if you're following the standard. I'd also double-check abbreviations vs standard English.